README.md
WPMU-network-ads
================

Wordpress Multisite Plugin, display ads code on every site on your network.

## Installation and Usage

### V1.0.X:
- Place all files on plugins directory
- Go to Network Settings->Network Ads Settings
- Select the code to display on every page and the websites that will 
have an exception

### V0.0.1: 
- Place the file in mu-plugins directory of your wpmu. 
- Edit adsense code accordingly. 
- Enjoy :)

## Future Features

- Internationalization

##Author

Boris Aramis Aguilar Rodriguez - `me <at> borisaguilar.com`

## License

GPLv2 License

## Changelog
### 0.0.1
- Hook on wp_foot to inject adsense code manually added on plugin
- Generate random class in HTML injected code to avoid not displaying 
ads by hiding with css.

### 1.0.1
- Admin Interface Ready
- Adsense code can be added and of course any custom you want :)
- removed px font size


CHANGES.md
# 0.0.4 Non-working version
- V0.0.1 is the only working version right now, please don't update!
- Working with Github-updater: https://github.com/afragen/github-updater


wp_foot_test.php
<?php
/**
 * @package NETWORK WP_FOOT
 * @version 0.1
 */
/*
Plugin Name: Network wp_foot
Plugin URI: http://etchq.net/
Description: on every page.
Author: Boris Aguilar & Jossemar Cordero
Version: 0.1
Author URI: http://ethcq.net/
*/

/** Step 1. */
function my_plugin_menu() {
	add_options_page( 'My Plugin Options', 'My Plugin', 'manage_options', 'my-unique-identifier', 'my_plugin_options' );
}

/** Step 2 (from text above). */
add_action( 'network_admin_menu', 'my_plugin_menu' );

/** Step 3. */
function my_plugin_options() {
	if ( !current_user_can( 'manage_options' ) )  {
		wp_die( __( 'You do not have sufficient permissions to access this page.' ) );
	}
	echo '<div class="wrap">';
	echo '<p>Here is where the form would go if I actually had options.</p>';
	echo '</div>';
}

function your_function() {
	$r = rand();
    	echo "<style>div.upstuff$r{text-align:center;font-size:120%;color:white;padding-top:10px;padding-bottom:10px;position: fixed;bottom: 0;left: 0;width: 100%;background-color: rgba(64,143,255,0.6);z-index: 999;}div.upstuff$r a{text-decoration:none;color:white;}</style><div class='upstuff$r'><!-- Banner Test --><div style='text-align:center;'><ins class='adsbygoogle' style='display:inline-block;width:468px;height:60px' data-ad-client='ca-pub-1130416518509640' data-ad-slot='6629883702'></ins></div><a href='http://etchq.net'><div>Creado Gratis en <span>etchq.net</span></div><div><strong>¡Crea tu sitio web con nosotros!</strong></div></a></div>";
    	echo '<script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script><script>(adsbygoogle = window.adsbygoogle || []).push({});</script>';
}
add_action('wp_footer', 'your_function');
?>


WPMU-Network-Ads.php
<?php
/**
 * The WordPress Plugin Boilerplate.
 *
 * A foundation off of which to build well-documented WordPress plugins that
 * also follow WordPress Coding Standards and PHP best practices.
 *
 * @package   WPMU Network Ads
 * @author    Boris Aguilar <me@borisguilar.com>
 * @license   GPL-2.0+
 * @link      https://github.com/boriscougar/WPMU-network-ads.git
 * @copyright 2013 Boris Aguilar
 *
 * @wordpress-plugin
 * Plugin Name:       WPMU Network Ads
 * Plugin URI:        https://github.com/boriscougar/WPMU-network-ads.git
 * Description:       Plugin to display Ads network-wide
 * Version:           1.0.1
 * Author:            Boris Aguilar
 * Author URI:        http://borisaguilar.com
 * Text Domain:       WPMU-network-ads
 * License:           GPL-2.0+
 * License URI:       http://www.gnu.org/licenses/gpl-2.0.txt
 * Domain Path:       /languages
 * GitHub Plugin URI: https://github.com/boriscougar/WPMU-network-ads
 */

// If this file is called directly, abort.
if ( ! defined( 'WPINC' ) ) {
	die;
}

/*----------------------------------------------------------------------------*
 * Public-Facing Functionality
 *----------------------------------------------------------------------------*/

/*
 *
 *
 */
require_once( plugin_dir_path( __FILE__ ) . '/public/class-wpmu-network-ads.php' );

/*
 * Register hooks that are fired when the plugin is activated or deactivated.
 * When the plugin is deleted, the uninstall.php file is loaded.
 *
 */
register_activation_hook( __FILE__, array( 'WPMUNetworkAds', 'activate' ) );
register_deactivation_hook( __FILE__, array( 'WPMUNetworkAds', 'deactivate' ) );

/*
 * TODO:
 *
 * - replace Plugin_Name_Admin with the name of the class defined in
 *   `class-plugin-name-admin.php`
 */
add_action( 'plugins_loaded', array( 'WPMUNetworkAds', 'get_instance' ) );

/*----------------------------------------------------------------------------*
 * Dashboard and Administrative Functionality
 *----------------------------------------------------------------------------*/

/*
 * TODO:
 *
 * If you want to include Ajax within the dashboard, change the following
 * conditional to:
 *
 * if ( is_admin() ) {
 *   ...
 * }
 *
 * The code below is intended to to give the lightest footprint possible.
 */
if ( is_admin() && ( ! defined( 'DOING_AJAX' ) || ! DOING_AJAX ) ) {

	require_once( plugin_dir_path( __FILE__ ) . '/admin/class-wpmunetworkads-admin.php' );
	add_action( 'plugins_loaded', array( 'WPMUNetworkAds_Admin', 'get_instance' ) );
}


admin/views/admin.php
<?php
/**
 * Represents the view for the administration dashboard.
 *
 * This includes the header, options, and other information that should provide
 * The User Interface to the end user.
 *
 * @package   WPMUNetworkAds
 * @author    Boris Aguilar <me@borisaguilar.com>
 * @license   GPL-2.0+
 * @link      http://example.com
 * @copyright 2013 Your Name or Company Name
 */
?>

<div class="wrap">

	<?php screen_icon(); ?>
	<h2><?php echo esc_html( get_admin_page_title() ); ?></h2>
	<form method="POST" action="/wp-admin/network/edit.php?action=wpmu_network_ads_update">
	      <label for="snippet-code"><h3>Snippet Code to use:</h3></label>
		<p><small>This snippet is shown on every page by hooking wp_foot.</small></p>
	      <textarea style="width:100%;" rows=10 id="snippet-code" name="snippet-code" rows=4 cols=2><?php echo(html_entity_decode(get_site_option('wpmu_network_snippet'))); ?></textarea>
		
	      <label for="snippet-code-exceptions"><h3>Excluded Sites:</h3></label>
		<p><small>One per line, don't include https or http just the URL. This URL's won't have ads.</small></p>
	      <textarea style="width:100%;" rows=10 id="snippet-code-exceptions" name="snippet-code-exceptions" rows=4 cols=2><?php echo(html_entity_decode(get_site_option('wpmu_network_exceptions'))); ?></textarea>
	      <?php submit_button(); ?>
	</form>

	<!-- TODO: Provide markup for your options page here. -->

</div>

admin/class-wpmunetworkads-admin.php
<?php
/**
 * Plugin Name.
 *
 * @package   WPMUNetworkAds_Admin
 * @author    Boris Aguilar <me@borisaguilar.com>
 * @license   GPL-2.0+
 * @link      http://borisaguilar.com
 * @copyright 2013 Boris Aguilar
 */

/**
 * Plugin class. This class should ideally be used to work with the
 * administrative side of the WordPress site.
 *
 * If you're interested in introducing public-facing
 * functionality, then refer to `class-plugin-name.php`
 *
 * TODO: Rename this class to a proper name for your plugin.
 *
 * @package WPMUNetworkAds_Admin
 * @author  Boris Aguilar <me@borisaguilar.com>
 */
class WPMUNetworkAds_Admin {

	/**
	 * Instance of this class.
	 *
	 * @since    0.0.2
	 *
	 * @var      object
	 */
	protected static $instance = null;

	/**
	 * Slug of the plugin screen.
	 *
	 * @since    0.0.2
	 *
	 * @var      string
	 */
	protected $plugin_screen_hook_suffix = null;

	/**
	 * Initialize the plugin by loading admin scripts & styles and adding a
	 * settings page and menu.
	 *
	 * @since     0.0.2
	 */
	private function __construct() {

		/*
		 * Call $plugin_slug from public plugin class.
		 */
		$plugin = WPMUNetworkAds::get_instance();
		$this->plugin_slug = $plugin->get_plugin_slug();

		// Load admin style sheet and JavaScript.
		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_styles' ) );
		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_scripts' ) );

		// Add the options page and menu item.
		add_action( 'network_admin_menu', array( $this, 'add_plugin_admin_menu' ) );

		// Add an action link pointing to the options page.
		$plugin_basename = plugin_basename( plugin_dir_path( __FILE__ ) . $this->plugin_slug . '.php' );
		add_filter( 'plugin_action_links_' . $plugin_basename, array( $this, 'add_action_links' ) );

		/*
		 * Define custom functionality.
		 *
		 * Read more about actions and filters:
		 * http://codex.wordpress.org/Plugin_API#Hooks.2C_Actions_and_Filters
		 */
		add_action( 'network_admin_edit_wpmu_network_ads_update', array( $this, 'save_wpmu_network_ads_snippet' ) );
		add_site_option('wpmu_network_snippet', '');
		add_site_option('wpmu_network_exceptions', '');
		add_filter( 'TODO', array( $this, 'filter_method_name' ) );

	}

	/**
	 * Return an instance of this class.
	 *
	 * @since     0.0.2
	 *
	 * @return    object    A single instance of this class.
	 */
	public static function get_instance() {

		// If the single instance hasn't been set, set it now.
		if ( null == self::$instance ) {
			self::$instance = new self;
		}

		return self::$instance;
	}

	/**
	 * Register and enqueue admin-specific style sheet.
	 *
	 * @since     0.0.2
	 *
	 * @return    null    Return early if no settings page is registered.
	 */
	public function enqueue_admin_styles() {

		if ( ! isset( $this->plugin_screen_hook_suffix ) ) {
			return;
		}

		$screen = get_current_screen();
		if ( $this->plugin_screen_hook_suffix == $screen->id ) {
			wp_enqueue_style( $this->plugin_slug .'-admin-styles', plugins_url( 'assets/css/admin.css', __FILE__ ), array(), WPMUNetworkAds::VERSION );
		}

	}

	/**
	 * Register and enqueue admin-specific JavaScript.
	 *
	 * @since     0.0.2
	 *
	 * @return    null    Return early if no settings page is registered.
	 */
	public function enqueue_admin_scripts() {

		if ( ! isset( $this->plugin_screen_hook_suffix ) ) {
			return;
		}

		$screen = get_current_screen();
		if ( $this->plugin_screen_hook_suffix == $screen->id ) {
			wp_enqueue_script( $this->plugin_slug . '-admin-script', plugins_url( 'assets/js/admin.js', __FILE__ ), array( 'jquery' ), WPMUNetworkAds::VERSION );
		}

	}

	/**
	 * Register the administration menu for this plugin into the WordPress Dashboard menu.
	 *
	 * @since    0.0.2
	 */
	public function add_plugin_admin_menu() {

		/*
		 * Add a settings page for this plugin to the Settings menu.
		 *
		 * NOTE:  Alternative menu locations are available via WordPress administration menu functions.
		 *
		 *        Administration Menus: http://codex.wordpress.org/Administration_Menus
		 *
		 * TODO:
		 *
		 * - Change 'Page Title' to the title of your plugin admin page
		 * - Change 'Menu Text' to the text for menu item for the plugin settings page
		 * - Change 'manage_options' to the capability you see fit
		 *   For reference: http://codex.wordpress.org/Roles_and_Capabilities
		 */
		$this->plugin_screen_hook_suffix = add_submenu_page(
			'settings.php',
			__( 'Network Ads Settings', $this->plugin_slug ),
			__( 'Network Ads Settings', $this->plugin_slug ),
			'manage_options',
			$this->plugin_slug,
			array( $this, 'display_plugin_admin_page' )
		);

	}

	/**
	 * Render the settings page for this plugin.
	 *
	 * @since    0.0.2
	 */
	public function display_plugin_admin_page() {
		include_once( 'views/admin.php' );
	}

	/**
	 * Add settings action link to the plugins page.
	 *
	 * @since    0.0.2
	 */
	public function add_action_links( $links ) {

		return array_merge(
			array(
				'settings' => '<a href="' . admin_url( 'options-general.php?page=' . $this->plugin_slug ) . '">' . __( 'Settings', $this->plugin_slug ) . '</a>'
			),
			$links
		);

	}

	/**
	 * NOTE:     Actions are points in the execution of a page or process
	 *           lifecycle that WordPress fires.
	 *
	 *           Actions:    http://codex.wordpress.org/Plugin_API#Actions
	 *           Reference:  http://codex.wordpress.org/Plugin_API/Action_Reference
	 *
	 * @since    0.0.5
	 */
	public function save_wpmu_network_ads_snippet() {
		$snippet_code = htmlentities(stripslashes($_POST['snippet-code']));
		$url_exceptions = $_POST['snippet-code-exceptions'];
		$url_exceptions = str_replace(" ", "", $url_exceptions);
		$url_exceptions = str_replace("\r\n", "\n", $url_exceptions);
		$url_exceptions = htmlentities(stripslashes($url_exceptions));
	       // i should save this somewhere
	       update_site_option('wpmu_network_snippet', $snippet_code);
	       update_site_option('wpmu_network_exceptions', $url_exceptions);
	       $url = (is_multisite()? network_admin_url('admin.php'): admin_url('admin.php'));
	       wp_redirect(
	           add_query_arg(
                       array( 'page' => $this->plugin_slug, 'updated' => 'true' ),
			   $url));
	}

	/**
	 * NOTE:     Filters are points of execution in which WordPress modifies data
	 *           before saving it or sending it to the browser.
	 *
	 *           Filters: http://codex.wordpress.org/Plugin_API#Filters
	 *           Reference:  http://codex.wordpress.org/Plugin_API/Filter_Reference
	 *
	 * @since    0.0.2
	 */
	public function filter_method_name() {
		// TODO: Define your filter hook callback here
	}

}

app/wp-content/plugins/WPMUNetworkAds/WPMUNetworkAds.php
<?php
/**
 * The WordPress Plugin Boilerplate.
 *
 * A foundation off of which to build well-documented WordPress plugins that
 * also follow WordPress Coding Standards and PHP best practices.
 *
 * @package   WPMU Network Ads
 * @author    Boris Aguilar <me@borisguilar.com>
 * @license   GPL-2.0+
 * @link      https://github.com/boriscougar/WPMU-network-ads.git
 * @copyright 2013 Boris Aguilar
 *
 * @wordpress-plugin
 * Plugin Name:       WPMU Network Ads
 * Plugin URI:        https://github.com/boriscougar/WPMU-network-ads.git
 * Description:       Plugin to display Ads network-wide
 * Version:           0.0.2
 * Author:            Boris Aguilar
 * Author URI:        http://borisaguilar.com
 * Text Domain:       plugin-name-locale
 * License:           GPL-2.0+
 * License URI:       http://www.gnu.org/licenses/gpl-2.0.txt
 * Domain Path:       /languages
 * GitHub Plugin URI: https://github.com/boriscougar/WPMU-network-ads.git
 */

// If this file is called directly, abort.
if ( ! defined( 'WPINC' ) ) {
	die;
}

/*----------------------------------------------------------------------------*
 * Public-Facing Functionality
 *----------------------------------------------------------------------------*/

/*
 * TODO:
 *
 *
 */
require_once( plugin_dir_path( __FILE__ ) . '/public/class-wpmu-network-ads.php' );

/*
 * Register hooks that are fired when the plugin is activated or deactivated.
 * When the plugin is deleted, the uninstall.php file is loaded.
 *
 */
register_activation_hook( __FILE__, array( 'WPMUNetworkAds', 'activate' ) );
register_deactivation_hook( __FILE__, array( 'WPMUNetworkAds', 'deactivate' ) );

/*
 * TODO:
 *
 * - replace Plugin_Name_Admin with the name of the class defined in
 *   `class-plugin-name-admin.php`
 */
add_action( 'plugins_loaded', array( 'WPMUNetworkAds', 'get_instance' ) );

/*----------------------------------------------------------------------------*
 * Dashboard and Administrative Functionality
 *----------------------------------------------------------------------------*/

/*
 * TODO:
 *
 * If you want to include Ajax within the dashboard, change the following
 * conditional to:
 *
 * if ( is_admin() ) {
 *   ...
 * }
 *
 * The code below is intended to to give the lightest footprint possible.
 */
if ( is_admin() && ( ! defined( 'DOING_AJAX' ) || ! DOING_AJAX ) ) {

	require_once( plugin_dir_path( __FILE__ ) . '/admin/class-wpmunetworkads-admin.php' );
	add_action( 'plugins_loaded', array( 'WPMUNetworkAds_Admin', 'get_instance' ) );

}

app/wp-content/plugins/WPMUNetworkAds/public/class-plugin-name.php
<?php
/**
 * WPMU Network Ads
 *
 * @package   WPMUNetworkAds
 * @author    Boris Aguilar <me@borisaguilar.com>
 * @license   GPL-2.0+
 * @link      http://borisaguilar.com
 * @copyright 2013 Boris Aguilar
 */

/**
 * Plugin class. This class should ideally be used to work with the
 * public-facing side of the WordPress site.
 *
 * If you're interested in introducing administrative or dashboard
 * functionality, then refer to `class-plugin-name-admin.php`
 *
 * TODO: Rename this class to a proper name for your plugin.
 *
 * @package WPMUNetworkAds
 * @author  Boris Aguilar <me@borisaguilar.com>
 */
class WPMUNetworkAds {

	/**
	 * Plugin version, used for cache-busting of style and script file references.
	 *
	 * @since   0.0.2
	 *
	 * @var     string
	 */
	const VERSION = '0.0.2';

	/**
	 *
	 * Unique identifier for your plugin.
	 *
	 * The variable name is used as the text domain when internationalizing strings
	 * of text. Its value should match the Text Domain file header in the main
	 * plugin file.
	 *
	 * @since    0.0.2
	 *
	 * @var      string
	 */
	protected $plugin_slug = 'wpmu-network-ads';

	/**
	 * Instance of this class.
	 *
	 * @since    0.0.2
	 *
	 * @var      object
	 */
	protected static $instance = null;

	/**
	 * Initialize the plugin by setting localization and loading public scripts
	 * and styles.
	 *
	 * @since     0.0.2
	 */
	private function __construct() {

		// Load plugin text domain
		add_action( 'init', array( $this, 'load_plugin_textdomain' ) );

		// Activate plugin when new blog is added
		add_action( 'wpmu_new_blog', array( $this, 'activate_new_site' ) );

		// Load public-facing style sheet and JavaScript.
		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_styles' ) );
		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );

		/* Define custom functionality.
		 * Refer To http://codex.wordpress.org/Plugin_API#Hooks.2C_Actions_and_Filters
		 */
		add_action( 'TODO', array( $this, 'action_method_name' ) );
		add_filter( 'TODO', array( $this, 'filter_method_name' ) );

	}

	/**
	 * Return the plugin slug.
	 *
	 * @since    0.0.2
	 *
	 *@return    Plugin slug variable.
	 */
	public function get_plugin_slug() {
		return $this->plugin_slug;
	}

	/**
	 * Return an instance of this class.
	 *
	 * @since     0.0.2
	 *
	 * @return    object    A single instance of this class.
	 */
	public static function get_instance() {

		// If the single instance hasn't been set, set it now.
		if ( null == self::$instance ) {
			self::$instance = new self;
		}

		return self::$instance;
	}

	/**
	 * Fired when the plugin is activated.
	 *
	 * @since    0.0.2
	 *
	 * @param    boolean    $network_wide    True if WPMU superadmin uses
	 *                                       "Network Activate" action, false if
	 *                                       WPMU is disabled or plugin is
	 *                                       activated on an individual blog.
	 */
	public static function activate( $network_wide ) {

		if ( function_exists( 'is_multisite' ) && is_multisite() ) {

			if ( $network_wide  ) {

				// Get all blog ids
				$blog_ids = self::get_blog_ids();

				foreach ( $blog_ids as $blog_id ) {

					switch_to_blog( $blog_id );
					self::single_activate();
				}

				restore_current_blog();

			} else {
				self::single_activate();
			}

		} else {
			self::single_activate();
		}

	}

	/**
	 * Fired when the plugin is deactivated.
	 *
	 * @since    0.0.2
	 *
	 * @param    boolean    $network_wide    True if WPMU superadmin uses
	 *                                       "Network Deactivate" action, false if
	 *                                       WPMU is disabled or plugin is
	 *                                       deactivated on an individual blog.
	 */
	public static function deactivate( $network_wide ) {

		if ( function_exists( 'is_multisite' ) && is_multisite() ) {

			if ( $network_wide ) {

				// Get all blog ids
				$blog_ids = self::get_blog_ids();

				foreach ( $blog_ids as $blog_id ) {

					switch_to_blog( $blog_id );
					self::single_deactivate();

				}

				restore_current_blog();

			} else {
				self::single_deactivate();
			}

		} else {
			self::single_deactivate();
		}

	}

	/**
	 * Fired when a new site is activated with a WPMU environment.
	 *
	 * @since    0.0.2
	 *
	 * @param    int    $blog_id    ID of the new blog.
	 */
	public function activate_new_site( $blog_id ) {

		if ( 1 !== did_action( 'wpmu_new_blog' ) ) {
			return;
		}

		switch_to_blog( $blog_id );
		self::single_activate();
		restore_current_blog();

	}

	/**
	 * Get all blog ids of blogs in the current network that are:
	 * - not archived
	 * - not spam
	 * - not deleted
	 *
	 * @since    0.0.2
	 *
	 * @return   array|false    The blog ids, false if no matches.
	 */
	private static function get_blog_ids() {

		global $wpdb;

		// get an array of blog ids
		$sql = "SELECT blog_id FROM $wpdb->blogs
			WHERE archived = '0' AND spam = '0'
			AND deleted = '0'";

		return $wpdb->get_col( $sql );

	}

	/**
	 * Fired for each blog when the plugin is activated.
	 *
	 * @since    0.0.2
	 */
	private static function single_activate() {
		// TODO: Define activation functionality here
	}

	/**
	 * Fired for each blog when the plugin is deactivated.
	 *
	 * @since    0.0.2
	 */
	private static function single_deactivate() {
		// TODO: Define deactivation functionality here
	}

	/**
	 * Load the plugin text domain for translation.
	 *
	 * @since    0.0.2
	 */
	public function load_plugin_textdomain() {

		$domain = $this->plugin_slug;
		$locale = apply_filters( 'plugin_locale', get_locale(), $domain );

		load_textdomain( $domain, trailingslashit( WP_LANG_DIR ) . $domain . '/' . $domain . '-' . $locale . '.mo' );
		load_plugin_textdomain( $domain, FALSE, basename( dirname( __FILE__ ) ) . '/languages/' );

	}

	/**
	 * Register and enqueue public-facing style sheet.
	 *
	 * @since    0.0.2
	 */
	public function enqueue_styles() {
		wp_enqueue_style( $this->plugin_slug . '-plugin-styles', plugins_url( 'assets/css/public.css', __FILE__ ), array(), self::VERSION );
	}

	/**
	 * Register and enqueues public-facing JavaScript files.
	 *
	 * @since    0.0.2
	 */
	public function enqueue_scripts() {
		wp_enqueue_script( $this->plugin_slug . '-plugin-script', plugins_url( 'assets/js/public.js', __FILE__ ), array( 'jquery' ), self::VERSION );
	}

	/**
	 * NOTE:  Actions are points in the execution of a page or process
	 *        lifecycle that WordPress fires.
	 *
	 *        Actions:    http://codex.wordpress.org/Plugin_API#Actions
	 *        Reference:  http://codex.wordpress.org/Plugin_API/Action_Reference
	 *
	 * @since    0.0.2
	 */
	public function action_method_name() {
		// TODO: Define your action hook callback here
	}

	/**
	 * NOTE:  Filters are points of execution in which WordPress modifies data
	 *        before saving it or sending it to the browser.
	 *
	 *        Filters: http://codex.wordpress.org/Plugin_API#Filters
	 *        Reference:  http://codex.wordpress.org/Plugin_API/Filter_Reference
	 *
	 * @since    0.0.2
	 */
	public function filter_method_name() {
		// TODO: Define your filter hook callback here
	}

}

public/class-wpmu-network-ads.php
<?php
/**
 * WPMU Network Ads
 *
 * @package   WPMUNetworkAds
 * @author    Boris Aguilar <me@borisaguilar.com>
 * @license   GPL-2.0+
 * @link      http://borisaguilar.com
 * @copyright 2013 Boris Aguilar
 */

/**
 * Plugin class. This class should ideally be used to work with the
 * public-facing side of the WordPress site.
 *
 * If you're interested in introducing administrative or dashboard
 * functionality, then refer to `class-plugin-name-admin.php`
 *
 * TODO: Rename this class to a proper name for your plugin.
 *
 * @package WPMUNetworkAds
 * @author  Boris Aguilar <me@borisaguilar.com>
 */
class WPMUNetworkAds {

	/**
	 * Plugin version, used for cache-busting of style and script file references.
	 *
	 * @since   0.0.2
	 *
	 * @var     string
	 */
	const VERSION = '0.0.2';

	/**
	 *
	 * Unique identifier for your plugin.
	 *
	 * The variable name is used as the text domain when internationalizing strings
	 * of text. Its value should match the Text Domain file header in the main
	 * plugin file.
	 *
	 * @since    0.0.2
	 *
	 * @var      string
	 */
	protected $plugin_slug = 'wpmu-network-ads';

	/**
	 * Instance of this class.
	 *
	 * @since    0.0.2
	 *
	 * @var      object
	 */
	protected static $instance = null;

	/**
	 * Initialize the plugin by setting localization and loading public scripts
	 * and styles.
	 *
	 * @since     0.0.2
	 */
	private function __construct() {

		// Load plugin text domain
		add_action( 'init', array( $this, 'load_plugin_textdomain' ) );

		// Activate plugin when new blog is added
		add_action( 'wpmu_new_blog', array( $this, 'activate_new_site' ) );

		// Load public-facing style sheet and JavaScript.
		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_styles' ) );
		add_action( 'wp_enqueue_scripts', array( $this, 'enqueue_scripts' ) );

		/* Define custom functionality.
		 * Refer To http://codex.wordpress.org/Plugin_API#Hooks.2C_Actions_and_Filters
		 */
		add_action( 'wp_footer', array( $this, 'custom_wp_foot' ) );
		add_filter( 'TODO', array( $this, 'filter_method_name' ) );

	}

	/**
	 * Return the plugin slug.
	 *
	 * @since    0.0.2
	 *
	 *@return    Plugin slug variable.
	 */
	public function get_plugin_slug() {
		return $this->plugin_slug;
	}

	/**
	 * Return an instance of this class.
	 *
	 * @since     0.0.2
	 *
	 * @return    object    A single instance of this class.
	 */
	public static function get_instance() {

		// If the single instance hasn't been set, set it now.
		if ( null == self::$instance ) {
			self::$instance = new self;
		}

		return self::$instance;
	}

	/**
	 * Fired when the plugin is activated.
	 *
	 * @since    0.0.2
	 *
	 * @param    boolean    $network_wide    True if WPMU superadmin uses
	 *                                       "Network Activate" action, false if
	 *                                       WPMU is disabled or plugin is
	 *                                       activated on an individual blog.
	 */
	public static function activate( $network_wide ) {

		if ( function_exists( 'is_multisite' ) && is_multisite() ) {

			if ( $network_wide  ) {

				// Get all blog ids
				$blog_ids = self::get_blog_ids();

				foreach ( $blog_ids as $blog_id ) {

					switch_to_blog( $blog_id );
					self::single_activate();
				}

				restore_current_blog();

			} else {
				self::single_activate();
			}

		} else {
			self::single_activate();
		}

	}

	/**
	 * Fired when the plugin is deactivated.
	 *
	 * @since    0.0.2
	 *
	 * @param    boolean    $network_wide    True if WPMU superadmin uses
	 *                                       "Network Deactivate" action, false if
	 *                                       WPMU is disabled or plugin is
	 *                                       deactivated on an individual blog.
	 */
	public static function deactivate( $network_wide ) {

		if ( function_exists( 'is_multisite' ) && is_multisite() ) {

			if ( $network_wide ) {

				// Get all blog ids
				$blog_ids = self::get_blog_ids();

				foreach ( $blog_ids as $blog_id ) {

					switch_to_blog( $blog_id );
					self::single_deactivate();

				}

				restore_current_blog();

			} else {
				self::single_deactivate();
			}

		} else {
			self::single_deactivate();
		}

	}

	/**
	 * Fired when a new site is activated with a WPMU environment.
	 *
	 * @since    0.0.2
	 *
	 * @param    int    $blog_id    ID of the new blog.
	 */
	public function activate_new_site( $blog_id ) {

		if ( 1 !== did_action( 'wpmu_new_blog' ) ) {
			return;
		}

		switch_to_blog( $blog_id );
		self::single_activate();
		restore_current_blog();

	}

	/**
	 * Get all blog ids of blogs in the current network that are:
	 * - not archived
	 * - not spam
	 * - not deleted
	 *
	 * @since    0.0.2
	 *
	 * @return   array|false    The blog ids, false if no matches.
	 */
	private static function get_blog_ids() {

		global $wpdb;

		// get an array of blog ids
		$sql = "SELECT blog_id FROM $wpdb->blogs
			WHERE archived = '0' AND spam = '0'
			AND deleted = '0'";

		return $wpdb->get_col( $sql );

	}

	/**
	 * Fired for each blog when the plugin is activated.
	 *
	 * @since    0.0.2
	 */
	private static function single_activate() {
		// TODO: Define activation functionality here
	}

	/**
	 * Fired for each blog when the plugin is deactivated.
	 *
	 * @since    0.0.2
	 */
	private static function single_deactivate() {
		// TODO: Define deactivation functionality here
	}

	/**
	 * Load the plugin text domain for translation.
	 *
	 * @since    0.0.2
	 */
	public function load_plugin_textdomain() {

		$domain = $this->plugin_slug;
		$locale = apply_filters( 'plugin_locale', get_locale(), $domain );

		load_textdomain( $domain, trailingslashit( WP_LANG_DIR ) . $domain . '/' . $domain . '-' . $locale . '.mo' );
		load_plugin_textdomain( $domain, FALSE, basename( dirname( __FILE__ ) ) . '/languages/' );

	}

	/**
	 * Register and enqueue public-facing style sheet.
	 *
	 * @since    0.0.2
	 */
	public function enqueue_styles() {
		wp_enqueue_style( $this->plugin_slug . '-plugin-styles', plugins_url( 'assets/css/public.css', __FILE__ ), array(), self::VERSION );
	}

	/**
	 * Register and enqueues public-facing JavaScript files.
	 *
	 * @since    0.0.2
	 */
	public function enqueue_scripts() {
		wp_enqueue_script( $this->plugin_slug . '-plugin-script', plugins_url( 'assets/js/public.js', __FILE__ ), array( 'jquery' ), self::VERSION );
	}

	/**
	 * NOTE:  Actions are points in the execution of a page or process
	 *        lifecycle that WordPress fires.
	 *
	 *        Actions:    http://codex.wordpress.org/Plugin_API#Actions
	 *        Reference:  http://codex.wordpress.org/Plugin_API/Action_Reference
	 *
	 * @since    0.0.4
	 */
	public function custom_wp_foot() {
		$except_sites = html_entity_decode(get_site_option('wpmu_network_exceptions'));
		$except_sites = explode("\n", $except_sites);
		$current_site = (string)$_SERVER["SERVER_NAME"];
		$site_found = in_array($current_site, $except_sites);
		if(!$site_found){
			$r = rand();
			echo "<style>div.upstuff$r{text-align:center;font-size:20px;line-height:20px;color:#ffffff;padding:10px 0 10px;position:fixed;bottom: 0;left: 0;right:0;background-color:rgba(64,143,255,0.6);z-index:9999;}div.upstuff$r a{text-decoration:none;color:#ffffff;}</style><div class='upstuff$r'><!-- Banner Start -->";
			echo html_entity_decode(get_site_option( 'wpmu_network_snippet' ));
			echo '</div>';
                }
	}

	/**
	 * NOTE:  Filters are points of execution in which WordPress modifies data
	 *        before saving it or sending it to the browser.
	 *
	 *        Filters: http://codex.wordpress.org/Plugin_API#Filters
	 *        Reference:  http://codex.wordpress.org/Plugin_API/Filter_Reference
	 *
	 * @since    0.0.2
	 */
	public function filter_method_name() {
		// TODO: Define your filter hook callback here
	}

}